src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -1005,7 +1007,7 @@ bool applyFunctionChainToColumn(
         if (func->getArgumentTypes().empty())
             return false;
 
-        auto argument_type = func->getArgumentTypes()[0];
+        auto argument_type = getArgumentTypeOfMonotonicFunction(*func);
```

Comment:
It might make sense to capture possible exceptions on `func->execute`, log them and continue the query without pruning.

Reply:
I would prefer to avoid shuch a logic here. It will hide the bug.
Overall, there is a lot of complicated logic in KeyCondition, mostly not skippable.